### PR TITLE
[GPU] Handle runtime scale value with scalar attention mask and scale input for SDPA

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - After this PR(https://github.com/openvinotoolkit/openvino/pull/31159) is applied, SDPA block of cross attention for Whisper decoder is fused. Accordingly, attention mask input is added as scalar value, but there is an issue that SDPA kernel does not properly handle the input argument index when the scale has input
 - Fix to handle runtime scale value with scalar attention mask and scale input for SDPA
 
#### The code and line that caused this issue (if it is not changed directly)
 - [sdpa_opt.cl#L145](https://github.com/openvinotoolkit/openvino/blob/81b12888af9ca39b18e1a13dca49428f9437acfd/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl#L145)
 - [sdpa_opt.cl#L728](https://github.com/openvinotoolkit/openvino/blob/81b12888af9ca39b18e1a13dca49428f9437acfd/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl#L728)
 - [sdpa_opt.cl#L844](https://github.com/openvinotoolkit/openvino/blob/81b12888af9ca39b18e1a13dca49428f9437acfd/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl#L844)
 ```
[GPU] ProgramBuilder build failed!
Program build failed(0_part_0):
1:3265:17: error: unknown type name 'INPUT4_TYPE'
 const __global INPUT4_TYPE* scale,
                ^
1:3402:10: warning: 'KEY_BLOCK_SIZE' macro redefined
....
1:7199:2: error: unknown type name 'INPUT4_TYPE'
 ATTN_SCALE_BUFFER_ARG
 ^
1:7184:49: note: expanded from macro 'ATTN_SCALE_BUFFER_ARG'
....
1:7287:17: error: unknown type name 'INPUT4_TYPE'
 const __global INPUT4_TYPE* scale,
                ^
1:7613:10: warning: 'KEY_BLOCK_READ' macro redefined
....
```

#### Reproduction step and snapshot (if applicable. Do not attach for customer model)
 - pytest test_ovc_mo.py --modules pipelines/production/onnx/pytorch_converted/heavy -k ONNX_WhisperDecoder --dynamism_type=None
 - ./benchmark_app -m /home/andrew/work/openvino/frameworks.ai.openvino.tests/e2e_oss/out_dir/ONNX_WhisperDecoder_api_2_True_batch_1_device_GPU_precision_FP16/model.xml -d GPU -hint none -niter 1 -nireq 1 -data_shape tokens[5,1],in_v_5a[5,51,512],in_v_4a[5,51,512],in_v_3a[5,51,512],in_v_2a[5,51,512],in_v_1a[5,51,512],in_v_0a[5,51,512],in_k_5a[5,51,512],in_k_4a[5,51,512],in_k_3a[5,51,512],in_k_2a[5,51,512],in_k_1a[5,51,512],in_k_0a[5,51,512],audio_features[5,1500,512]

#### Problematic graph
 - image

#### Checklist
 - [x] Is it a proper fix? (not a workaround)
 - [x] Did you include test case for this fix, if necessary?
 - [x] Did you review existing test that can be extended to cover this scenario? Which test did you review?
[smoke_ScaledAttnDynamic4D_GPU](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/functional/single_layer_tests/dynamic/scaled_dot_product_attention.cpp)

### Tickets:
 - 170070

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl [hunk 1]
```diff
@@ -725,7 +733,11 @@ KERNEL(sdpa_opt)(
 
 #if HAS_SCALE_INPUT
     #define ATTN_SCALE_BUFFER , scale
+#if HAS_ATTN_MASK_INPUT
     #define ATTN_SCALE_BUFFER_ARG , const __global INPUT4_TYPE* scale
+#else
+    #define ATTN_SCALE_BUFFER_ARG , const __global INPUT3_TYPE* scale
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl]: not using SCALE_TYPE?

Comment (andrew-k-park) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl]: apply it and merge duplicated definitions

**Review by yeonbok — APPROVED** [review]
